hazelcast-hibernate/hazelcast-hibernate4/src/main/java/com/hazelcast/hibernate/HazelcastCacheRegionFactory.java

```diff
@@ -58,6 +55,13 @@ public EntityRegion buildEntityRegion(final String regionName, final Properties
                 new IMapRegionCache(regionName, instance, properties, metadata));
     }
 
+    public NaturalIdRegion buildNaturalIdRegion(final String regionName, final Properties properties,
+                                                final CacheDataDescription metadata)
+            throws CacheException {
+        return new HazelcastNaturalIdRegion<IMapRegionCache>(instance, regionName, properties, metadata,
+                new IMapRegionCache(regionName, instance, properties, null));
```

Comment:
Unlike in `buildEntityRegion`, we need to pass in `null` for `metadata` here, as the only thing `IMapRegionCache` uses this for is to extract the version comparator, and natural id regions are never versioned.